schema/representations.md

```diff
@@ -0,0 +1,257 @@
+
```

Comment:
This is interesting information.  I'm liking some of principles in discussions.  As a follower of the project, I have some suggested adds.  It's a little hard to understand at first.  A few improvements to suggest:

 * A Background section with criteria and purpose to motivate.  For example, "IPFS applications like domain-specific applications and generic data-crawl software will traverse a graph of data found in IPFS.  They will discover sub-graphs and identify the type of schema-definitions found.  If they support,  they parse the data and continue with data traversal.  If they don't support the schema, they still identify the schema format that's not supported. Lookup service to find support?  Given a CID referencing a data block, a schema-aware IPFS client can identify the schema type from data in the block, schema for that data block, and show the found data".  And so.

 * What are evaluation criteria - moving some of the discussed principles into the document.  

 * Certain use-cases providing design purposes.  Encoding filesystem-mounting semantics?  Broad vision.  Examples of mount filesystems from IPFS/IPLD data: read-only, read-write, cache hints - specific, not generic.  IMDB data might have good examples too.  Put concrete examples in an introductory section as backstory, then reference them inline, not using "Fizzlebop".  If no Filecoin references, I'm sad.

 * Strategy for forward-compatibility: Are multiformats or in-block schema-CID refs planned?  Will a block of data be self-describing?  Will you later support new use-cases and introduce new schema capabilities?  How?  Will IPLD schema be open, so that non-PL contributors will be able to define IPLD data and software plugins for today's  encodings & data schemas?  How can metadata be encoded so that current and future software are strong with such multi-schemas?  Will they follow all data links because CBOR?  If not, Unsupported section can be useful.  Links to other documents can organize some of info that now exists.

 * How is avoided https://www.xkcd.com/927/ ?  JSON-Schema, protobuf and Avro should be mentioned.  What good or bad points, and how proposal takes their best points to improve within the IPFS ecosystem.  Can IPLD support data using these encodings?  Or unsupported? Help others know what it is/isn't and how to get there from here.

Keep the good work guys.

Reply:
:thumbsup: I think the path to doing some of this is to just get this merged and then start to fill out these pieces. This is a good TODO list though.